plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/proxy/NugetFeedFetcher.java

```diff
@@ -114,6 +97,35 @@ public Integer cachePackageFeed(final Repository proxy, final URI nugetQuery,
   }
 
   /**
+   * Consume a page of feed XML from the InputStream.
+   *
+   * @return The URI for the next (unvisited) page, if there is one.
+   */
+  private URI parseFeed(final InputStream is, final FeedSplicer splicer, final Set<String> visited,
+                        final boolean followNextPageLinks) throws IOException
+  {
+    try {
+      final String nextPageUrl = splicer.consumePage(is);
+
+      if (followNextPageLinks && nextPageUrl != null) {
+        if (visited.add(nextPageUrl)) {
+          return new URI(nextPageUrl);
+        }
+        else {
+          log.warn("Page cycle detected: {} -> {}", visited, nextPageUrl);
+        }
+      }
+      return null;
+    }
+    catch (XmlPullParserException e) {
+      throw new IOException("Invalid nuget feed XML received.", e);
```

Comment:
have to remember log messages are not proper sentences, thus the ending "." is odd, its also not consistent with the log right below it.  Suggest, and have so for a while, that punctuation like this at the end of a log be avoided